### DISCUSSION TITLE
Allow custom text selection actions in Android 11 and up

### DISCUSSION THREAD

**Discussion by scubajeff** [community/community#17495] [Mobile]
After version 11, Android changes package visibility behavior, to allow other apps, like translator, wikipedia, to add custom actions when user select text in Github Mobile, please add the following to it's manifest:

```
<queries>
 <intent>
   <action android:name="android.intent.action.PROCESS_TEXT" />
   <data android:mimeType="text/plain" />
 </intent>
</queries>

```
Refer to https://developer.android.com/training/package-visibility/use-cases#custom-text-selection-actions

**Comment by HenSquared** [comment]
Hi @scubajeff,

Thank you for the feature request and for the sample code. I've opened an Issue internally for where the team will discuss this request further.

**Comment by alcere** [comment]
Hi @scubajeff,

Thank you for reaching out! 

Could you please provide more context about the issue? Some steps explaining what you currently see after selecting some text and what you would like to see will be helpful for us.

Thanks in advance!

**Comment by scubajeff** [comment]
Let's say you are reading an issue comment like this, and assume you're not so fluent in English reading and you find a word difficult to understand, you can select the word, now github app will display a context menu with standard action like copy, cut, etc., if github mobile allow a translator app to shown it's action in this context menu, then you can easily select that action to help you with the translation. Wikipedia app also provide an action call "Search in Wikipedia" too.

**Reply by alcere** [reply]
Thanks for the detailed description! Is this issue reproducible from a particular screen, or is it something that happens everywhere within the app?

**Reply by scubajeff** [reply]
This is a feature request, not a bug report.

**Reply by alcere** [reply]
Sure! and we appreciate that! The current behaviour we see running the Android app v1.67.1, on Android 11 and 12, is exactly the flow described above having the Wikipedia or Google Translate app installed, therefore we are a bit confused and considering the idea that you might be experiencing a bug for a particular screen.

**Reply by scubajeff** [reply]
My bad! Wikipedia does show up. But how about this one: https://play.google.com/store/apps/details?id=site.leos.setter

**Reply by scubajeff** [reply]
So the fact is Github mobile only allow certain apps tapping into this context menu.

**Reply by alcere** [reply]
Thanks for the help! we will continue investigating the issue now that we have more information.
